protocol/messenger_handler.go

```diff
@@ -837,7 +837,7 @@ func (m *Messenger) HandleContactUpdate(state *ReceivedMessageState, message pro
 		chat.LastClockValue = message.Clock
 	}
 
-	if contact.ContactRequestState == ContactRequestStateMutual {
+	if contact.ContactRequestState == ContactRequestStateMutual && chat.DeletedAtClockValue < message.Clock {
```

Comment:
It would be nice to update test(s) to cover this particular condition.

Reply:
Yep, it would be. Sadly though there are no tests to update, and I'm not into writing new ones for this particular if clause.